lib/mpifx_comm.fpp

```diff
@@ -38,28 +43,46 @@ contains
   !! \param error  Error flag on return containing the first error occuring
   !!     during the calls mpi_comm_size and mpi_comm_rank.
   !!
-  subroutine mpifx_comm_init(self, commid, error)
+  subroutine mpifx_comm_init_comm(self, comm, error)
     class(mpifx_comm), intent(out) :: self
-    integer, intent(in), optional :: commid
+    type(mpi_comm), intent(in), optional :: comm
     integer, intent(out), optional :: error
 
     integer :: error0
 
-    call getoptarg(MPI_COMM_WORLD, self%id, commid)
-    call mpi_comm_size(self%id, self%size, error0)
+    if (present(comm)) then
+      self%comm = comm
+    else
+      self%comm = MPI_COMM_WORLD
+    end if
+    self%id = self%comm%mpi_val
+    call mpi_comm_size(self%comm, self%size, error0)
     call handle_errorflag(error0, "mpi_comm_size() in mpifx_comm_init()", error)
     if (error0 /= 0) then
       return
     end if
-    call mpi_comm_rank(self%id, self%rank, error0)
+    call mpi_comm_rank(self%comm, self%rank, error0)
     call handle_errorflag(error0, "mpi_comm_rank() in mpifx_comm_init()", error)
     if (error0 /= 0) then
       return
     end if
     self%leadrank = 0
     self%lead = (self%rank == self%leadrank)
 
-  end subroutine mpifx_comm_init
+  end subroutine mpifx_comm_init_comm
+
+
+  subroutine mpifx_comm_init_int(self, commid, error)
```

Comment:
Can you please add comments for the routine, but also why is is called _init_int() ? This is not very informative.
```suggestion
  !> Initializes a communicator from a numerical id.
  !!
  !! \param self  Initialized instance on exit.
  !! \param commid  Numerical MPI Communicator ID
  !! \param error  Error flag on return containing the first error occurring
  !!     during the calls mpi_comm_size and mpi_comm_rank.
  !!
  subroutine mpifx_comm_from_id(self, commid, error)
```